src/Apia/Defaults.hs

```diff
@@ -0,0 +1,365 @@
+      Just val → opts { optTime = val }
+      _        → opts
+
+setUnprovenNoError ∷ Options → Config → Options
+setUnprovenNoError opts config = setVerbose newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "unproven-conjecture-no-error" config of
+      Just val → opts { optUnprovenNoError = val }
+      _        → opts
+
+setVerbose ∷ Options → Config → Options
+setVerbose opts config = setWithCVC4 newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "verbose" config of
+      Just val  → case verboseOpt val opts of
+                        Right o → o
+                        Left _  → opts
+      _           → opts
+
+setWithCVC4 ∷ Options → Config → Options
+setWithCVC4 opts config = setWithE newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-cvc4" config of
+      Just val → opts { optWithCVC4 = val }
+      _        → opts
+
+setWithE ∷ Options → Config → Options
+setWithE opts config = setWithEquinox newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-e" config of
+      Just val → opts { optWithE = val }
+      _        → opts
+
+setWithEquinox ∷ Options → Config → Options
+setWithEquinox opts config = setWithIleanCoP newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-equinox" config of
+      Just val → opts { optWithEquinox = val }
+      _        → opts
+
+setWithIleanCoP ∷ Options → Config → Options
+setWithIleanCoP opts config = setWithMetis newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-ileancop" config of
+      Just val → opts { optWithIleanCoP = val }
+      _        → opts
+
+setWithMetis ∷ Options → Config → Options
+setWithMetis opts config = setWithSPASS newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-metis" config of
+      Just val → opts { optWithMetis = val }
+      _        → opts
+
+setWithSPASS ∷ Options → Config → Options
+setWithSPASS opts config = setWithtptp4X newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-spass" config of
+      Just val → opts { optWithSPASS = val }
+      _        → opts
+
+setWithtptp4X ∷ Options → Config → Options
+setWithtptp4X opts config = setWithVampire newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-tptp4X" config of
+      Just val → opts { optWithtptp4X = val }
+      _        → opts
+
+setWithVampire ∷ Options → Config → Options
+setWithVampire opts config = setWithZ3 newOpts config
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-vampire" config of
+      Just val → opts { optWithVampire = val }
+      _        → opts
+
+setWithZ3 ∷ Options → Config → Options
+setWithZ3 opts config = newOpts
+  where
+    newOpts ∷ Options
+    newOpts = case lookConfig "with-z3" config of
+      Just val → opts { optWithZ3 = val }
+      _        → opts
+
```

Comment:
Is it strictly necessary that each `setFoo` function (except the "first" one) depends of the previous one?